src/components/fields/file-upload/file-upload.tsx

```diff
@@ -234,7 +243,7 @@ export const FileUploadInner = (props: IGenericFieldProps<IFileUploadSchema>) =>
 				onChange={handleChange}
 				onDelete={handleDelete}
 				title={renderHtmlText(label)}
-				warning={renderHtmlText(warning)}
+				warning={renderHtmlText(warning || schemaWarning)}
```

Comment:
yeah this looks like an edge case, at this point, I will say the onus is on the dev using this library - he/she should manage the warnings and show both messages if intended.

perhaps there may be a need to show multiple alerts but without a use case, it is too early to assume so we can stick to the current approach to avoid over-engineering it.